syntax = "proto3";
package bcos.protocol;
message PBRawTransactionMetaData
{
    bytes hash = 1;
    bytes to = 2;
    uint32 attribute = 3;
    string source = 4;
};
// Mainly used for block data packet encoding when network transmission
message PBRawBlock{
    int32 version = 1;
    bytes header = 2;
    repeated bytes transactions = 3;
    repeated bytes receipts = 4;
    // In some scenarios, only transaction hashes are included in the block
    repeated PBRawTransactionMetaData transactionsMetaData = 5;
    repeated bytes receiptsHash = 6;
    // the block type, include: 
    // CompleteBlock, OnlyWithTransactionsHash
    int32 type = 7;
    repeated bytes nonceList = 8;
}
